util_file.py

```diff
@@ -93,8 +93,38 @@ def ls_remote(user, machine, directory, port=22):
 
     command = f"ls {directory}"
     stdin, stdout, stderr = ssh.exec_command(command)
+    stdout = stdout.read().decode(errors='ignore'), stderr.read().decode(errors='ignore')
 
-    return stdout.read().decode(errors='ignore'), stderr.read().decode(errors='ignore')
+    return stdout
+
+
+def find_remote(user, machine, directory, serach, port=22):
+    '''
+    Perform 'find {directory} -name '{custom_search}'"' over ssh onto linux machine.
+    Note this returns the full file path, not relative to 'directory'.
+
+    Parameters
+    ----------
+    user : str
+    machine : str
+        IP address of the maching you will connect to.
+    directory : str
+    search : str
+    port : int, default 22
+
+    Returns
+    -------
+    str
+    '''
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh.connect(machine, username=user, port=port)
+
+    command = f"find {directory} -name '{serach}'"
```

Comment:
should this be custom_search, instead of serach?

Reply:
Yes maybe - in the code we always use the word "serach", but in the first line of the docstring it says "custom_search" - in the variables in the docstring it actually says "search"... I'll change all to "search", which I assume it what my brain intended.